> Can you rebase ?

done

> IMO those backtips are weird, and I don't see how they improve the output.
> They are also an oddity if you compare to other software that output colors, such as `git`.

The backticks are used as a kind of quotation mark. GCC has very similar error messages, just with different quote characters. For example, here is an error from DMD (with `-color=off`):

    test.d(2): Error: function `test.f(int* x)` is not callable using argument types `(double)`
    test.d(2):        cannot pass argument `1.2` of type `double` to parameter `int* x`

And here is GCC's (shortened):

    test.c:2:16: error: incompatible type for argument 1 of 'f'
    test.c:1:13: note: expected 'int *' but argument is of type 'double'

GCC also uses color highlighting for the same things as DMD: location info, "error:", quoted code. GCC prints the quotation marks even when colors are off.

> Would you be okay with removing them when color are off ?

I'm against it. There's a reason why people on the forum put code snippets in backticks even though we have no color highlighting there. It makes for better readability. Silly example:

    test.d(1): Error: undefined identifier sequence

Newbie: "What's an 'identifier sequence'? Is that like an alias sequence or what?"

    test.d(1): Error: undefined identifier `sequence`

Newbie: "Oh!"

--- a/src/dmd/errors.d
+++ b/src/dmd/errors.d
@@ -662,7 +662,6 @@ private void colorSyntaxHighlight(ref OutBuffer buf)
                     immutable pre = "";
                     i = buf.insert(iCodeStart, pre);
                     i = buf.insert(i, codebuf[]);
-                    i--; // point to the ending ) so when the for loop does i++, it will see the next character
                     break;
                 }
                 inBacktick = true;

--- a/test/compilable/testcolor.sh
+++ b/test/compilable/testcolor.sh
@@ -35,6 +35,12 @@ check -color=auto "test" "$expectedWithoutColor"
 check -color=on "test" "$expectedWithColor"
 check -color=off "test" "$expectedWithoutColor"
 
+gooCode="void foo() {} void main() { goo(); }"
+gooExpectedWithoutColor='__stdin.d(1): Error: undefined identifier `goo`, did you mean function `foo`?'
+gooExpectedWithColor=$'\033[1m__stdin.d(1): \033[1;31mError: \033[mundefined identifier `\033[0;36m\033[m\033[1mgoo\033[0;36m\033[m`, did you mean function `\033[0;36m\033[m\033[1mfoo\033[0;36m\033[m`?'
+check -color=on "$gooCode" "$gooExpectedWithColor"
+check -color=off "$gooCode" "$gooExpectedWithoutColor"
+
 if [[ "$(script --version)" == script\ from\ util-linux\ * ]]
 then
     actual="$(SHELL="$(command -v bash)" TERM="faketerm" script -q -c "echo test | ( $DMD -c -o- -)" /dev/null | normalize)" || true

